radix-engine/src/kernel/track.rs

```diff
@@ -992,3 +989,14 @@ impl<'a, 'b> BalanceChangeAccounting<'a, 'b> {
             .substate
     }
 }
+
+/// Removes the `left.intersection(right)` from both `left` and `right`, in place, without
+/// computing (or allocating) the intersection itself.
+/// Implementation note: since Rust has no "iterator with delete" capabilities, the implementation
+/// uses a (normally frowned-upon) side-effect of a lambda inside `.retain()`.
+/// Performance note: since the `BTreeSet`s are inherently sorted, the implementation _could_ have
+/// an `O(n+m)` runtime (i.e. traversing 2 iterators). However, it would then contain significantly
+/// more bugs than the `O(n * log(m))` one-liner below.
+fn remove_intersection<T: Ord>(left: &mut BTreeSet<T>, right: &mut BTreeSet<T>) {
+    left.retain(|id| !right.remove(id));
```

Comment:
_Note from author:_

Still worth reviewing! :p

Reply:
As you said, merge-sort would be optimal. 

A minor improvement to the one-liner would be to compare the size before `retain`.